src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloader.java

```diff
@@ -136,4 +150,21 @@ public void download(
       throw exception;
     }
   }
+
+  public List<URL> getMirroredUrls(List<URL> urls) throws IOException {
+    List<URL> mirroredUrls = new ArrayList<>(urls.size());
+    for (URL url : urls) {
+      String portStr = "";
+      if ( url.getPort() != -1 ) {
+        portStr = String.format(":%d", url.getPort());
+      }
+      String mirroredUrlStr = mirror.toString().replaceAll("/*$","") + "/" + url.getHost() + portStr + "/" + url.getPath().replaceAll("^/*","");
```

Comment:
Could you extract this URL transformation into a method and add a simple unit test for it that ensures it does what we think it does? ;)

Reply:
Thanks for your review, updated, PTAL:)